core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -78,7 +78,15 @@ object UniqueTriggerActivation {
                 return true
             }
             OneTimeFreeUnitRuins -> {
-                val unit = civInfo.getEquivalentUnit(unique.params[0])
+                var unit = civInfo.getEquivalentUnit(unique.params[0])
+                if ( unit.hasUnique(UniqueType.FoundCity) && civInfo.isOneCityChallenger()) {
+                    if (ruleSet.units[Constants.worker] != null) {
+                        unit = civInfo.getEquivalentUnit(Constants.worker)
```

Comment:
You're still using hardcoded unit names, please search for units by type and unique (like you did to see if it's a settler!)